apps/profiles/serializers/profiles.py

```diff
@@ -38,10 +44,17 @@ class Meta:
             "email",
             "image_url",
             "is_email_verified",
+            "city",
         )
 
     def update(self, instance, validated_data):
         if 'image_url' in validated_data and not validated_data['image_url']:
             delete_image_if_exists(instance)
 
+        if "city" in validated_data:
+            new_city = validate_city(validated_data["city"])
+            if instance.city != new_city:
+                instance.city = new_city
+                instance.save()
+
         return super().update(instance, validated_data)
```

Comment:
Let's stick with a new approach - removing logic from serializers and moving logic to views. Just override perform_update function in the profile viewset and make this logic there